### DISCUSSION TITLE
How to treat the contents of an USB-C device within the mediaplayer.

### DISCUSSION THREAD

**Discussion by agdejager** [MoleMan1024/audiowagon#32] [Ideas]
With all due respect, i think there is a fundamental wrong way to handle the FAT32 USB-C devices within the mediaplayer..

If there is a clear directory structure as given by Mr. Moleman, then the way it is treated until now is very complex and not even errorprone. To index the data the use of the metadata within files is being used (not even errorprone with regular well known albums) and is very slow.

I would rather index the directory/file structure itself as is being done in for instance in a Skoda Citigo-e iV where a flashcard with the same structure as my USB sticks is recognized within tens of a second and is playable at once. There you can scroll trough all directory levels and files directly. It is only depending on the directore/filestructure. How many levels should not be an issue and every music file is a playable item within the given directory path. It is up to the user how that is structured, but of course for the logic use a certain directory structure is advisable. 

As a proof of my thinking i will attach my last logfile of my only well known albums USB device. 

[audiowagon_2021-10-07_10-25-51.zip](https://github.com/MoleMan1024/audiowagon/files/7300887/audiowagon_2021-10-07_10-25-51.zip)

This file shows even missing files from well known albums as well as many albums for every artist on well known composition albums. 

If following a directory/file structure the handling would be faster, more errorprone, more feasible and usable and also more flexible imho.

with many regards and greetings, abraham

**Comment by MoleMan1024** [comment]
There is a feature request [here](https://github.com/MoleMan1024/audiowagon/issues/26) already to allow for file/directory browsing. If I added a setting to turn indexing off (if files/directories are enough for you), then I think this would do all you want, it would not need indexing and you could navigate by file/directory immediately. You could only ever search by directory or filename however, so this is only for people who have a media library with "clean" filenames.

But yes I agree this would be very useful to have for people who have lots of files without metadata (e.g. from YouTube).

**Reply by agdejager** [reply]
Ok, that is certainly enough for me, because i use clean and logical directory structures and therefore i do not need extra metadata indexing.
If you can add that feature, i would be very happy.

with many greetings and regards, Abraham.
